src/components/sideBarContentEdit/SelectField.tsx

```diff
@@ -293,3 +305,85 @@ export const SingleSelectField = ({
     </Select>
   );
 };
+
+// function to return dir based on the currentPath
+
+function findAndReturnDirs(
+  objArray: IDir[],
+  path: string[],
+  depth: number,
+  index?: number
+): IDir[] {
+  if (depth === 0) {
+    return []; // Return null if the desired depth is reached, but the path is not found.
+  }
+
+  if (index === 1) {
+    return objArray;
+  }
+
+  for (const obj of objArray) {
+    if (obj.slug === path[depth - 2]) {
+      if (path.length === 2 && depth === 2) {
```

Comment:
Can you put a comment on why we're doing this before the recursion? Its entirely clear

Reply:
> Can you put a comment on why we're doing this before the recursion? Its entirely clear
for line 327? 
reason for doing , we want to check the current length of the array for example advancing-bitcoin/2019 would be of length 2,  and if path if found we return the nestDir or an empty array if it's not two we slice it again